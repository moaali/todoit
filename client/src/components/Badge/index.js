import React from 'react'
import PropTypes from 'prop-types'

import './index.scss'

const Badge = ({ className, content, ...restProps }) => (
  <span
    className={`badge ${className}`}
    {...restProps}
  >
    {content}
  </span>
);

Badge.defaultProps = {
  className: '',
}

Badge.propTypes = {
  className: PropTypes.string,
  content: PropTypes.oneOfType([
    PropTypes.number,
    PropTypes.string,
  ]).isRequired,
}

export default Badge;
