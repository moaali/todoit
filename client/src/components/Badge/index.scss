@import 'settings/index';
@import 'tools/index';
@import 'helpers/index';

.badge {
  // Typography
  font-size: 12px;
  font-weight: 600;
  text-align: center;
  line-height: 1;

  // Box Model
  display: inline-block;
  padding: 3px 5px;
  min-width: 24px;

  // Cosmetics
  border-radius: 10em;
  background-color: rgba($color-pivotal, 0.05);
  color: rgba($color-pivotal, 0.7);
}

.badge--primary {
  // Cosmetics
  background-color: rgba($color-primary, 0.1);
  color: $color-primary;
}

.badge--success {
  // Cosmetics
  background-color: rgba($color-success, 0.1);
  color: $color-success;
}

.badge--warning {
  // Cosmetics
  background-color: rgba($color-warning, 0.1);
  color: $color-warning;
}
