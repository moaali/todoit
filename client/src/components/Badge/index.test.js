import React from 'react'

import Badge from './'

describe('Badge', () => {
  it('should match snapshot', () => {
    const badgeElement = shallow(<Badge content={1} />);
    expect(toJson(badgeElement)).toMatchSnapshot();
  })
})
