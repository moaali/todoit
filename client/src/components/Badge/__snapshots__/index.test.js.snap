// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge should match snapshot 1`] = `
<span
  className="badge "
>
  1
</span>
`;
